SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestEntityInteration/EntityInteractionTest.cpp

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "EntityInteractionTest.h"
+#include "EntityInteractionTestActor.h"
+#include "Kismet/GameplayStatics.h"
+
+#include "SpatialGDKSettings.h"
+
+ASpatialEntityInteractionTest::ASpatialEntityInteractionTest()
+	: Super()
+{
+	Author = "Nicolas";
+	Description = TEXT("");
+}
+
+#define EXPECT_IMMEDIATE(Actor, FunctionName, Context)                                                                                     \
+	do                                                                                                                                     \
+	{                                                                                                                                      \
+		FString DebugString("Immediate execution of a ");                                                                                  \
+		DebugString += FunctionName;                                                                                                       \
+		DebugString += " on ";                                                                                                             \
+		DebugString += Actor->GetName();                                                                                                   \
+		DebugString += " : ";                                                                                                              \
+		DebugString += Context;                                                                                                            \
```

Comment:
I don't know what UE's "printf" variant is, but use string formatting to construct this? Would make this easier to read, and potentially also save you a whole bunch of lines.

(same below)